src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py

```diff
@@ -119,16 +119,18 @@
 register_cli_argument('lock', 'ids', nargs='+', options_list=('--ids'), help='One or more resource IDs (space delimited). If provided, no other "Resource Id" arguments should be specified.')
 
 register_cli_argument('managedapp', 'resource_group_name', arg_type=resource_group_name_type, help='the resource group of the managed application', id_part='resource_group')
-register_cli_argument('managedapp', 'appliance_name', options_list=('--name', '-n'), id_part='name')
+register_cli_argument('managedapp', 'application_name', options_list=('--name', '-n'), id_part='name')
 
 register_cli_argument('managedapp definition', 'resource_group_name', arg_type=resource_group_name_type, help='the resource group of the managed application definition', id_part='resource_group')
-register_cli_argument('managedapp definition', 'appliance_definition_name', options_list=('--name', '-n'), id_part='name')
+register_cli_argument('managedapp definition', 'application_definition_name', options_list=('--name', '-n'), id_part='name')
 
 register_cli_argument('managedapp create', 'name', options_list=('--name', '-n'), help='name of the new managed application', completer=None)
 register_cli_argument('managedapp create', 'location', help='the managed application location')
 register_cli_argument('managedapp create', 'managedapp_definition_id', options_list=('--managedapp-definition-id', '-d'), help='the full qualified managed application definition id')
 register_cli_argument('managedapp create', 'managedby_resource_group_id', options_list=('--managed-rg-id', '-m'), help='the resource group managed by the managed application')
 register_cli_argument('managedapp create', 'parameters', help='JSON formatted string or a path to a file with such content', type=file_type)
 
-register_cli_argument('managedapp definition create', 'lock_level', **enum_choice_list(ApplianceLockLevel))
+register_cli_argument('managedapp definition create', 'lock_level', **enum_choice_list(ApplicationLockLevel))
 register_cli_argument('managedapp definition create', 'authorizations', options_list=('--authorizations', '-a'), nargs='+', help="space separated authorization pairs in a format of <principalId>:<roleDefinitionId>")
+register_cli_argument('managedapp definition create', 'createUiDefinition', options_list=('--create-ui-def', '-c'), help='JSON formatted string or a path to a file with such content', type=file_type)
+register_cli_argument('managedapp definition create', 'mainTemplate', options_list=('--main-template', '-t'), help='JSON formatted string or a path to a file with such content', type=file_type)
```

Comment:
Does this actually work?  It should by in python_case, not camelCase. @lmazuel's recent changes to serialization/deserialization might mean that this "just works"...